### DISCUSSION TITLE
What does `install.yaml` do that `flux install` doesn't?

### DISCUSSION THREAD

**Discussion by vanakema** [fluxcd/flux2#4214] [Q&A] [answered]
The command `flux install --export > test2.yaml` results in the same output that `flux boostrap`ing puts in the `gotk-components.yaml` file. This file has about 2000 lines less than `install.yaml` from the equivalent flux release. We tried diffing the files, but there are such significant differences that it wasn't helpful (potentially ordering differences of resources causing such a large diff).

Can someone explain why these two files are so dramatically different? 

Our goal is to use cdk8s to build our flux system configuration and push the synthed artifacts in an OCI repository, by first doing a manual "apply" of the install manifests, but we're not sure which one makes most sense to use.

**Comment by stefanprodan** [comment, marked as answer]
The `install.yaml` from the release page is generated like this: https://github.com/fluxcd/flux2/blob/47c8e5f44d8e45844ae4016d89eec10c121fa7c9/.github/workflows/release.yaml#L137-L139

> Our goal is to use cdk8s to build our flux system configuration and push the synthed artifacts in an OCI repository

An OCI artifact with the Flux manifests is published here https://github.com/fluxcd/flux2/pkgs/container/flux-manifests

To configure Flux to update itself from the official OCI repo see this example: https://github.com/stefanprodan/flux-local-dev/tree/main/kubernetes/clusters/local/flux-system
